src/crimson/common/interruptible_future.h

```diff
@@ -1416,42 +1410,45 @@ struct interruptor
 	    typename Result = futurize_t<std::invoke_result_t<Func>>>
   static inline Result async(Func&& func) {
     return seastar::async([func=std::forward<Func>(func),
-			   interrupt_condition=interrupt_cond<InterruptCond>]() mutable {
+			   interrupt_condition=interrupt_cond<InterruptCond>.interrupt_cond]
+			  () mutable {
       return non_futurized_call_with_interruption(
 	  interrupt_condition, std::forward<Func>(func));
     });
   }
 
   static void yield() {
-    ceph_assert(interrupt_cond<InterruptCond>);
-    auto interruption_condition = interrupt_cond<InterruptCond>;
-    interrupt_cond<InterruptCond>.release();
+    ceph_assert(interrupt_cond<InterruptCond>.interrupt_cond);
+    auto interruption_condition = interrupt_cond<InterruptCond>.interrupt_cond;
     INTR_FUT_DEBUG(
-      "interruptible_future_detail::yield() yielding out, interrupt_cond {},{} cleared",
+      "interruptible_future_detail::yield() yielding out, "
+      "interrupt_cond {},{} cleared",
       (void*)interruption_condition.get(),
       typeid(InterruptCond).name());
+    interrupt_cond<InterruptCond>.reset();
```

Comment:
In the commit message:

```
Ccurrently, interrupt conditionss ...
```
Currently, interrupt conditions ...

```
1.if a seastar::get/yield() inside the passed lambda, the interrupt_cond should be erased when it
is yielded back;
2.otherwise, the interrupt_cond should be erased.
```
IIUC: if condition1., the interrupt_cond should ^not be erased when it is yielded ~~back~~;
Am I correct?

Reply:
Sorry for the typo, repushed.